src/SuperVault.sol

```diff
@@ -684,54 +796,43 @@ contract SuperVault is BaseStrategy, ISuperVault {
 
         /// @dev update SV data
         weights = newWeights;
-        superformIds = finalSuperformIds;
+        superformIds = finalSuperformIds_;
         numberOfSuperforms = length;
 
-        emit RebalanceComplete(finalSuperformIds, newWeights);
+        emit RebalanceComplete(finalSuperformIds_, newWeights);
     }
 
     /// @notice Changes the whitelist for a Superform ID
-    /// @param superformId The Superform ID to change
-    /// @param isWhitelisted Whether to whitelist or blacklist
-    function _changeSuperformWhitelist(uint256 superformId, bool isWhitelisted) internal {
-        bool currentlyWhitelisted = whitelistedSuperformIds[superformId];
+    /// @param superformId_ The Superform ID to change
+    /// @param isWhitelisted_ Whether to whitelist or blacklist
+    function _changeSuperformWhitelist(uint256 superformId_, bool isWhitelisted_) internal {
+        bool currentlyWhitelisted = whitelistedSuperformIds[superformId_];
 
         // Only process if there's an actual change
-        if (currentlyWhitelisted != isWhitelisted) {
-            whitelistedSuperformIds[superformId] = isWhitelisted;
+        if (currentlyWhitelisted != isWhitelisted_) {
+            whitelistedSuperformIds[superformId_] = isWhitelisted_;
 
-            if (isWhitelisted) {
-                _addToWhitelist(superformId);
+            if (isWhitelisted_) {
+                _addToWhitelist(superformId_);
             } else {
-                _removeFromWhitelist(superformId);
+                _removeFromWhitelist(superformId_);
             }
 
-            emit SuperformWhitelisted(superformId, isWhitelisted);
+            emit SuperformWhitelisted(superformId_, isWhitelisted_);
         }
     }
 
     /// @notice Adds a superform ID to the whitelist array
     /// @param superformId The Superform ID to add
     function _addToWhitelist(uint256 superformId) internal {
         whitelistedSuperformIds[superformId] = true;
-        whitelistedSuperformIdArray.push(superformId);
+        whitelistedSuperformIdsSet.add(superformId);
```

Comment:
The extra mapping `whitelistedSuperformIds[]` isn't necessary, because this is can be checked in the set via `contains()`.

Reply:
Fixed in [251f26f](https://github.com/superform-xyz/SuperVaults/pull/36/commits/251f26f4ef39b7f99084c971336b75b9cfc143c6)